lib/experimental/full-page-client-side-navigation.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Registers full page client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+// Set the navigation mode to full page client-side navigation.
+wp_interactivity_config( 'core/router', array( 'navigationMode' => 'fullPage' ) );
+
+// Enqueue the interactivity router script.
+function _gutenberg_enqueue_interactivity_router() {
+	wp_enqueue_script_module( '@wordpress/interactivity-router' );
+}
+
+add_action( 'init', '_gutenberg_enqueue_interactivity_router', 20 );
+
+// Add directives to all links.
+// This should probably be done per site, not by default when this option is enabled.
+function _gutenberg_add_enhanced_pagination( $parsed_block ) {
+	if ( 'core/query' !== $parsed_block['blockName'] ) {
+		return $parsed_block;
+	}
+
+	$parsed_block['attrs']['enhancedPagination'] = true;
+	return $parsed_block;
+}
+
+add_filter( 'render_block_data', '_gutenberg_add_enhanced_pagination', 10 );
```

Comment:
Nitpick: 10 is the default priority, so it can be omitted.

Reply:
Done as part of [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/36e090173ee052f9d5bbb8a92eca3304e2df2bcb).